src/delay.ml

```diff
@@ -307,6 +307,74 @@ let parse_durationl l =
   let s = String.concat " " l in
   parse_duration s
 
+let duration_of_daypack_duration (x : Daypack_lib.Duration.t) : duration =
+  C.Period.lmake ~day:x.days ~hour:x.hours ~minute:x.minutes ~second:x.seconds ()
+
+let get_current_tz_offset_s () =
+  Ptime_clock.current_tz_offset_s () |> Option.get |> Option.some
```

Comment:
> Why use ptime here ? We can already get the local timezone offset from calendarlib.

Use of Ptime was closer to consistency than very immediate reason: some time zone offsets are at increments of smaller than an hour, in which case I have no clue what calendarlib does exactly. There is otherwise no reason why calendarlib cannot be used.

> Also, all the calculations inside fugit are done in GMT, and only converted at the boundary, which I find much simpler to understand (and resilient to TZ switch ...)

The time zone offset is largely required by the search function, as the search would make more sense over the local time than GMT time - when we say "Thursday 9am" we probably want the local time Thursday 9am than GMT time Thursday 9am. The returned result by the search function is in UTC/GMT in any case (seconds since Unix epoch).

Reply:
> some time zone offsets are at increments of smaller than an hour, in which case I have no clue what calendarlib does exactly

Actually, I do not either .. should probably check.

> The time zone offset is largely required by the search function, as the search would make more sense over the local time

Ah, this does make sense!